### PR TITLE
CU/6027-Dylan-UpdateReactDevtools

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Updated React Devtools to the latest version to remove some vulnerabilities, also updated the electron resolution to remove a vulnerability and removed an unused resolution that wasn't needed, will fix jest one when the testing libraries are updated again

## Screenshots/Video
Screenshot 2024-01-05 at 11 02 43 AM

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
Definition of done:

 react-devtools is upgrade to current version
 patch is either removed or revised
 Yarn resolutions have been either updated or remove
 Security vulnerabilities are gone or reduced as far as possible (yarn audit)

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/package.json [hunk 1]
```diff
@@ -37,11 +37,6 @@
     "@department-of-veterans-affairs/mobile-component-library": "0.0.22",
     "@expo/react-native-action-sheet": "^3.13.0",
     "@react-native-async-storage/async-storage": "^1.18.1",
-    "@react-native-firebase/analytics": "^14.5.0",
-    "@react-native-firebase/app": "^14.5.0",
-    "@react-native-firebase/crashlytics": "^14.5.0",
-    "@react-native-firebase/perf": "^14.5.0",
-    "@react-native-firebase/remote-config": "14.5.0",
```
Comment (Sparowhawk) [inline VAMobile/package.json:44]: not sure how these got in here, but I removed the extra declarations

File: VAMobile/package.json [hunk 2]
```diff
@@ -147,10 +142,9 @@
     "babel-plugin-module-resolver/**/json5": "^1.0.2",
     "detox/**/minimatch": "^3.0.5",
     "detox/**/minimist": "^1.2.6",
-    "prettier": "^2.8.8",
     "react-devtools/**/got": "^11.8.5",
-    "react-devtools/**/minimist": "^1.2.6",
-    "react-devtools/electron": "^25.0.1",
+    "react-devtools/electron": "^25.8.1",
```
Comment (Sparowhawk) [inline VAMobile/package.json:146]: the version that was patched for this was 25.8.1 not 25.0.1

File: VAMobile/ios/Pods/RCT-Folly/folly/portability/Time.h-e [hunk 3]
```diff
@@ -30,7 +30,7 @@
 #if __MACH__ &&                                                       \
         ((!defined(TARGET_OS_OSX) || TARGET_OS_OSX) &&                \
          (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_12)) || \
-    (TARGET_OS_IPHONE && (__IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_10_0))
+    (TARGET_OS_IPHONE)
```
Comment (alexandec) [inline VAMobile/ios/Pods/RCT-Folly/folly/portability/Time.h-e:33]: Any potential issues with removing this version check from the pod?

**Review by alexandec — APPROVED** [review]
This branch runs fine for me locally, and I was able to open devtools. Just added one question above

**Review by alexandec — APPROVED** [review]

**Pull request merged**